### PR TITLE
Budget force dataset dependencies

### PR DESCRIPTION
**Pull request by selflein**

### CODE REVIEW

File: src/llamafactory/train/sft/metric.py [hunk 1]
```diff
@@ -125,6 +125,38 @@ def eval_logit_processor(logits: "torch.Tensor", labels: "torch.Tensor") -> "tor
...
+        if res is not None:
+            res = res.group(grp_ind).strip().lower()
+            
+            if res.endswith("."):
+                res = res[:-1]
+            if res.endswith(")") and res.startswith("("):
+                res = res[1:-1]
+            return res
+        
+    return text.lower()
```
Comment (selflein) [inline src/llamafactory/train/sft/metric.py:158]: Factoring this out of the class is the main change.

File: scripts/vllm_infer_for_data_gen.py [hunk 2]
```diff
@@ -49,66 +48,62 @@ def __init__(self, force_thinking, force_wait, n_reflections, tokenizer, samplin
...
+            for p in pred:
+                suppress_end_of_thinking = p.split(self.think_end_token)[0].strip()
+                wait_text = self.npr.choice(self.wait_tokens)
+                new_prompt = prompt + suppress_end_of_thinking + f" {wait_text} "
+                new_propmt_token_ids = self.tokenizer.encode(new_prompt)
+                new_inp = deepcopy(inp)
+                new_inp["prompt_token_ids"] = new_propmt_token_ids
+                new_inputs.append(new_inp)
+
+        return {"prompts": new_inputs, "sampling_params": wait_sampling_params}
```
Comment (selflein) [inline scripts/vllm_infer_for_data_gen.py:82]: These changes are just stylistically. It returns early to reduce nesting
